### PR TITLE
[MI-2571]: Done the review fixes of PR: #284

### PR DESCRIPTION
**Pull request by Nityanand13**
[PR #284](https://github.com/mattermost/mattermost-plugin-zoom/pull/284)

### CODE REVIEW

File: server/http.go [hunk 1]
```diff
@@ -28,8 +28,13 @@ const (
 	zoomOAuthUserStateLength = 4
 	settingDataError         = "something went wrong while getting settings data"
 	askForPMIMeeting         = "Would you like to use your personal meeting ID?"
-	apiToUpdatePMI           = "/plugins/%s/api/v1/updatePMI"
-	apiToAskForPMI           = "/plugins/%s/api/v1/askPMI"
+	pathWebhook              = "/webhook"
+	pathStartMeeting         = "/api/v1/meetings"
```
Comment (Kshitij-Katiyar) [inline server/http.go:32]: APIs starting with path ```/api/v1/``` have been added by you or did they exist before hand as well?

File: server/http.go [hunk 2]
```diff
@@ -402,7 +410,7 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, webh
 
 	post, appErr := p.API.GetPost(postID)
 	if appErr != nil {
-		p.API.LogWarn("Could not get the meeting post by id", "Error", appErr.Error())
+		p.API.LogWarn("Could not get the meeting post by id", "err", appErr.Error())
```
Comment (Kshitij-Katiyar) [inline server/http.go]: Make this similar to above "failed to get the meeting post by id"

File: server/http.go [hunk 3]
```diff
@@ -700,38 +706,44 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 	var createMeetingErr error
 	userPMISettingPref, err := p.getPMISettingData(user.Id)
 	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
```
Comment (Kshitij-Katiyar) [inline server/http.go]: are u sure that this is should "http.StatusUnauthorized"

Comment (Nityanand13) [reply server/http.go]: Oh sorry it should be `StatusInternalServerError`.

File: server/http.go (hunk 1 above)
Comment (Nityanand13) [reply server/http.go:32]: It exists earlier as well https://github.com/mattermost/mattermost-plugin-zoom/blob/8e898623befc730f7a161ff15a60a8ffe5e28433/server/http.go#L48

File: server/http.go (hunk 2 above)
Comment (Nityanand13) [reply server/http.go]: Although I did it, Is there any reason behind that?

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/command.go [hunk 4]
```diff
@@ -155,15 +155,16 @@ func (p *Plugin) runStartCommand(args *model.CommandArgs, user *model.User, topi
 
 	userPMISettingPref, err := p.getPMISettingData(user.Id)
 	if err != nil {
-		p.askPreferenceForMeeting(user.Id, args.ChannelId)
-		return "", nil
+		return "", err
 	}
 
+	isCreateMeetingWithPMI := false
```
Comment (manojmalik20) [inline server/command.go]:
```suggestion
	createMeetingWithPMI := false
```

File: server/http.go [hunk 5]
```diff
@@ -130,7 +135,9 @@ func (p *Plugin) startMeeting(action string, userID string, channelID string) {
 
 	var meetingID int
 	var createMeetingErr error
+	isCreateMeetingWithPMI := false
```
Comment (manojmalik20) [inline server/http.go]: Same.

File: server/http.go [hunk 6]
```diff
@@ -201,7 +209,7 @@ func (p *Plugin) setPMI(w http.ResponseWriter, r *http.Request) {
 	p.API.UpdateEphemeralPost(mattermostUserID, post)
 	w.Header().Set("Content-Type", "application/json")
 	if err := json.NewEncoder(w).Encode(post); err != nil {
-		http.Error(w, "failed to write the response", http.StatusInternalServerError)
+		p.API.LogWarn("failed to write response", "Error", err.Error())
```
Comment (manojmalik20) [inline server/http.go]:
```suggestion
		p.API.LogError("failed to write response", "Error", err.Error())
```

File: server/http.go [hunk 7]
```diff
@@ -318,13 +326,13 @@ func (p *Plugin) completeUserOAuthToZoom(w http.ResponseWriter, r *http.Request)
 	} else {
 		meeting, err := client.CreateMeeting(zoomUser, defaultMeetingTopic)
 		if err != nil {
-			p.API.LogWarn("Error creating the meeting", "Error", err.Error())
+			p.API.LogWarn("Error creating the meeting", "err", err.Error())
```
Comment (manojmalik20) [inline server/http.go]:
```suggestion
			p.API.LogWarn("Error creating the meeting", "error", err.Error())
```

File: server/http.go [hunk 8]
```diff
@@ -366,7 +374,7 @@ func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
 
 	b, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		p.API.LogWarn("Cannot read the body from Webhook")
+		p.API.LogWarn("failed to read the body from Webhook", "Error", err.Error())
```
Comment (manojmalik20) [inline server/http.go]:
```suggestion
		p.API.LogWarn("failed to read the body from Webhook", "error", err.Error())
```

File: server/http.go [hunk 9]
```diff
@@ -402,7 +410,7 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, webh
 
 	post, appErr := p.API.GetPost(postID)
 	if appErr != nil {
-		p.API.LogWarn("Could not get the meeting post by id", "Error", appErr.Error())
+		p.API.LogWarn("failed to get the meeting post by id", "err", appErr.Error())
```
Comment (manojmalik20) [inline server/http.go]:
```suggestion
		p.API.LogWarn("failed to get the meeting post by id", "error", appErr.Error())
```

File: server/http.go [hunk 10]
```diff
@@ -435,26 +443,26 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, webh
 	post.Props["meeting_status"] = zoom.WebhookStatusEnded
 	post.Props["attachments"] = []*model.SlackAttachment{&slackAttachment}
 
-	_, appErr = p.API.UpdatePost(post)
-	if appErr != nil {
-		p.API.LogWarn("Could not update the post", "Error", appErr.Error())
+	if _, appErr = p.API.UpdatePost(post); appErr != nil {
+		p.API.LogWarn("Could not update the post", "err", appErr.Error())
```
Comment (manojmalik20) [inline server/http.go]:
```suggestion
		p.API.LogWarn("Could not update the post", "error", appErr.Error())
```

File: server/http.go [hunk 11]
```diff
@@ -435,26 +443,26 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, webh
...
 	if appErr = p.deleteMeetingPostID(meetingPostID); appErr != nil {
-		p.API.LogWarn("failed to delete db entry", "Error", appErr.Error())
+		p.API.LogWarn("failed to delete db entry", "error", appErr.Error())
 		return
 	}
 
 	w.Header().Set("Content-Type", "application/json")
 	if err := json.NewEncoder(w).Encode(post); err != nil {
+		p.API.LogWarn("failed to write response", "error", err.Error())
 		http.Error(w, "failed to write the response", http.StatusInternalServerError)
```
Comment (manojmalik20) [inline server/http.go]: If you are keeping this line here, why did you remove it above?

File: server/http.go [hunk 12]
```diff
@@ -700,38 +706,44 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 	var createMeetingErr error
 	userPMISettingPref, err := p.getPMISettingData(user.Id)
 	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
 		p.askPreferenceForMeeting(user.Id, req.ChannelID)
 		return
 	}
 
+	isCreateMeetingWithPMI := false
```
Comment (manojmalik20) [inline server/http.go]: Same.

File: server/telemetry.go [hunk 13]
```diff
@@ -29,6 +29,12 @@ func (p *Plugin) trackMeetingStart(userID, source string) {
 	})
 }
 
+func (p *Plugin) trackMeetingStatus(userID string, isCreateMeetingWithPMI bool) {
```
Comment (manojmalik20) [inline server/telemetry.go]:
```suggestion
func (p *Plugin) trackMeetingType(userID string, usePMI bool) {
```

File: server/telemetry.go [hunk 14]
```diff
@@ -29,6 +29,12 @@ func (p *Plugin) trackMeetingStart(userID, source string) {
...
+	_ = p.tracker.TrackUserEvent("meeting_status", userID, map[string]interface{}{
+		"isCreateMeetingWithPMI": isCreateMeetingWithPMI,
```
Comment (manojmalik20) [inline server/telemetry.go]:
```suggestion
	_ = p.tracker.TrackUserEvent("meeting_type", userID, map[string]interface{}{
		"use_pmi": usePMI,
```

File: server/http.go (hunk 11 above)
Comment (Nityanand13) [reply server/http.go]: Now removed this from here as well.
https://github.com/mattermost/mattermost-plugin-zoom/pull/284/files/f6c187931258d07dbe04cdfbbb883e94f866ba1b#r1061019522

**Review by manojmalik20 — APPROVED** [review]

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]
